zerver/lib/webhooks/common.py

```diff
@@ -114,3 +130,38 @@ def validate_extract_webhook_http_header(request: HttpRequest, header: str,
         raise MissingHTTPEventHeader(header)
 
     return extracted_header
+
+class AuthStrategies:
+    SIMPLE_TOKEN = 1
+
+def http_api_callback(bot: UserProfile, auth_strategy: int,
+                      callback_url: str) -> Optional[Dict[str, Any]]:
+    """
+        The way we contact the callback endpoint will vary based on the
+        authentication method required to contact it.
+
+        For SIMPLE_TOKEN based auth, we can simply modify the url of the
+        endpoint accordingly.
+    """
+    try:
+        if int(auth_strategy) == AuthStrategies.SIMPLE_TOKEN:
+            bot_configuration = {config.key: config.value
+                                 for config in BotConfigData.objects.filter(bot_profile=bot)}
+            details = {
+                "callback_url": callback_url,
+                "auth_param": bot_configuration["callback_auth_param"],
+                "auth_token": bot_configuration["callback_auth_token"],
+            }
+            complete_url = "{callback_url}?{auth_param}={auth_token}".format(**details)
+            response = requests.get(complete_url)
+            if response.status_code != 200:
+                raise UnsuccessfulResponseError(bot, callback_url, response.status_code)
+            else:
+                return json.loads(response.text)
+        else:
+            raise ValueError("Unsupported auth_strategy.")
+    except KeyError:
+        # If the any of the data is missing then we should assume that the user did not want to
+        # provide their API keys and then webhook view using this method should recognize this and
+        # act accordingly.
+        return None
```

Comment:
Should we be returning `None` here? If any of the data is missing, is there a way to notify the user as to what is missing?

Reply:
This is something that might require further discussion. My focuses here were towards backwards compatibility and making this feature as optional as possible (please refer to Rishi Gupta's comment on issue #10907).

I wanted to make it such that people currently using a stripe integration wouldn't have to deal with an error message upon updating their version of Zulip. Also someone setting up a stripe integration would not *have* to open advanced settings to explicitly mention that they don't want to provide their API keys.

One thing I'm curious about now is how would we modify the docs to let people know that such a feature exists now (maybe we could directly modify the doc for the integration itself)?

Reply:
@timabbott could you please weigh in on this issue as well?

Reply:
Okay, I may be missing something here because I'm having trouble grasping how this maintains backwards compatibility. I think it would be super helpful if you could please add a commit on top of this where an actual webhook uses this feature, so that we can review and test the code with a running third-party API! :)

Reply:
Sounds like a good idea. I'll try adding it to the stripe integration.